### ISSUE TITLE
LiveView Native

### ISSUE THREAD

**Issue by nelsonic** [dwyl/learn-elixir#177]
ElixirConf 2022 - Brian Cardarella - What is LiveView Native?
https://youtu.be/dnDGh_Jmw-s
<img width="814" alt="image" src="https://user-images.githubusercontent.com/194400/189217135-8ad8c6d7-989e-4ba1-adcc-138965b37daa.png">

This looks promising. 🤞

**Comment by nelsonic** [comment]
WatchOS and MacOS are apparently possible with [**`SwiftUI`**](https://developer.apple.com/xcode/swiftui/) 😮 
<img width="810" alt="image" src="https://user-images.githubusercontent.com/194400/189227063-2ac9d8cc-a92e-460f-ae5e-9489f9800be8.png">

**Comment by nelsonic** [comment]
https://native.live/
https://github.com/liveviewnative/ios-tutorial

**Comment by nelsonic** [comment]
Roadmap:
<img width="810" alt="image" src="https://user-images.githubusercontent.com/194400/189229068-3678ce28-eabc-4b3c-8069-82154462b63c.png">

**Comment by ndrean** [comment]
Seems like 80% of the front code is still Native to me, isn't it?  I didn't really understand with they didn't target Reactive Native: plenty of people use it, it does the job, in JS,  for both major platforms, and you can [transform a HTML string into React elements](https://github.com/remarkablemark/html-react-parser).

**Comment by nelsonic** [comment]
@ndrean the _reason_ for not targeting RN is simple: it's _sloooooow_. 🐌 

For _simple_ Apps where responsiveness (App booting/loading time) is not a priority, RN is perfectly fine.

We recently worked on an (_massive + existing_) RN App for a large fintech client, 
it had over **500+ distinct screens** ... (***don't ask ...*** 🙄)
and _everything_ about the **Developer Experience** was inferior to writing **`Swift`** ("_real_" Native) code. Debugging is still _painful_ and app reloading is still clunky in 2022!
Don't even get me started on the **dependencies** in RN! 😢 
The time to setup a new dev machine was almost an hour. ⏳ 

We know the appeal of RN very well: write `JS` code and get a "Native" App ... 💭 
This is the _theory_ and it _mostly_ works ...
But for any App that grows to do anything other than basic "CRUD" it gets _painfully_ slow.
We're talking _several_ seconds for a "cold start" of the App on the _latest_ iPhones! ⏳ 
like go grab a coffee while the app boots ... ☕ 
I could go on about the RN DevEx (for bigger apps and larger teams) at length ... 🙊 


_Why_ handicap `LiveView Native` by targeting RN? Because it has more packages? [**`YAGNI!`**](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

**`SwiftUI`** is made by **`Apple`**:  https://developer.apple.com/xcode/swiftui/
![image](https://user-images.githubusercontent.com/194400/189436554-56a17f2a-c76b-4a4d-8e06-07217e6be244.png)

In other words it's the _best_ UX/DevEx you can get on iOS/Mac/iPad _and_ ***`WatchOS`***! 🚀 

Yes, there might (_superficially_) be fewer components/packages in **`SwiftUI`** than **RN**,
but that's a massive [red herring](https://en.wikipedia.org/wiki/Red_herring) argument.
Focus on what _really_ matters to the people using your App; the UX/performance!
Building with **`SwiftUI`** is a **no-brainer**.
Consider the happiness of the Dev/Engineer _building_ the app **`SwiftUI`**, it's _even_ more of a **no-brainer**!

There is zero compromise with this approach. 
We could not be more delighted that Brian & Co took it! ❤️ 

If they can pull off the Android with **`Jetpack`** https://developer.android.com/jetpack 
![image](https://user-images.githubusercontent.com/194400/189436993-358e640d-c467-442f-80c6-e207e2e4019e.png)

Then "full stack" `Elixir` could become a real thing! 😍 🤞 
It remains to be seen how easy it will be to ***`debug`*** ... 💭 
But we are _very_ optimistic. 🙏

**Comment by ndrean** [comment]
Yes, probably RN is slow, even Flutter is slow compared to real native. I have only one experience with a very small app. My concern is that Elxiir/Liveview seems to contribute to only say 10% of the code base so you still need to be an IOS-Swift or Android-Compose expert to develop something. I plan to learn Swift to make a small mobile app (a hobby): I like kitesurfing, especially downwinds, but I struggle to find people to organize it (the problem with cars).  I want to make a simple app where you declare an even: "when", "from where" and "to where" you would like to do your downwind, and let people join the list. Extra tools would be notifications and sending mails. You can see the events whether as a list or on a map with two markers and a line. Just 3 screens, login, list, and map and a few modals. I started to do this as a web app but I want to go mobile. Let's see!
